pkg/reconciler/reconciler.go

```diff
@@ -682,13 +691,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Re
 	// This is a safety measure to ensure that the chart is fully uninstalled before the CR is deleted.
 	if obj.GetDeletionTimestamp() == nil && !controllerutil.ContainsFinalizer(obj, uninstallFinalizer) {
 		log.V(1).Info("Adding uninstall finalizer.")
+		patch := client.MergeFrom(obj.DeepCopy())
 		obj.SetFinalizers(append(obj.GetFinalizers(), uninstallFinalizer))
-		if err := r.client.Update(ctx, obj); err != nil {
+		if err = r.client.Patch(ctx, obj, patch); err != nil {
```

Comment:
I'd suggest keeping the current scope of `err` for the sake of changing as little as possible.
```suggestion
		if err := r.client.Patch(ctx, obj, patch); err != nil {
```

Reply:
done